animations/lib/src/misc/curved_animation.dart

```diff
@@ -0,0 +1,156 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'package:flutter/material.dart';
+
+class CurvedAnimationDemo extends StatefulWidget {
+  static const String routeName = '/misc/curved_animation';
+
+  @override
+  _CurvedAnimationDemoState createState() => _CurvedAnimationDemoState();
+}
+
+class CurveChoice {
+  final Curve curve;
+  final String name;
+  const CurveChoice({this.curve, this.name});
+}
+
+class _CurvedAnimationDemoState extends State<CurvedAnimationDemo>
+    with SingleTickerProviderStateMixin {
+  AnimationController controller;
+  Animation<double> animationRotation;
+  Animation<Offset> animationTranslation;
+  static const _duration = Duration(seconds: 4);
+  List<CurveChoice> curves = [
+    CurveChoice(curve: Curves.bounceIn, name: 'Bounce In'),
+    CurveChoice(curve: Curves.easeInCubic, name: 'Ease In Cubic'),
+    CurveChoice(curve: Curves.easeInExpo, name: 'Ease In Expo'),
+    CurveChoice(curve: Curves.elasticIn, name: 'Elastic In'),
+    CurveChoice(curve: Curves.easeInQuart, name: 'Ease In Quart'),
+    CurveChoice(curve: Curves.easeInCirc, name: 'Ease In Circle'),
+  ];
+  CurveChoice selectedForwardCurve, selectedReverseCurve;
+  CurvedAnimation curvedAnimation;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = AnimationController(
+      duration: _duration,
+      vsync: this,
+    );
+    selectedForwardCurve = curves[0];
+    selectedReverseCurve = curves[0];
+    curvedAnimation = CurvedAnimation(
+      parent: controller,
+      curve: selectedForwardCurve.curve,
+      reverseCurve: selectedReverseCurve.curve,
+    );
+    animationRotation = Tween<double>(
+      begin: 0,
+      end: 2 * math.pi,
+    ).animate(curvedAnimation)
+      ..addListener(() {
+        setState(() {});
+      })
+      ..addStatusListener((status) {
+        if (status == AnimationStatus.completed) {
+          controller.reverse();
+        }
+      });
+    animationTranslation = Tween<Offset>(
+      begin: Offset(-1, 0),
+      end: Offset(1, 0),
+    ).animate(curvedAnimation)
+      ..addListener(() {
+        setState(() {});
+      })
+      ..addStatusListener((status) {
+        if (status == AnimationStatus.completed) {
+          controller.reverse();
+        }
+      });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: Text('Curved Animation')),
```

Comment:
nit: Add a comma in between the two parentheses at the end of this line:

```
appBar: AppBar(title: Text('Curved Animation'),),
```

While it's not completely necessary, it'll space the parent/child out the same way it's done in most of our other samples.